src/main/java/controller/EditRuleController.java

```diff
@@ -78,40 +89,88 @@ private void onSaveRule() {
 			return;
 		}
 
-		String newRule = getJavascriptString(rawRuleConditions);
-
-		// gets the rules list from the main controller
-		ArrayList<CodeQualityRule> rulesList = mainC.getRulesList();
-
-		if (!rulesList.contains(rule)) {
-			rule = new CodeQualityRule(newName, newRule, false, editRulePopup.isAdvancedMode());
-		} else {
-			rulesList.remove(rule);
-
-			if (rule.isDefault()) {
-				rule = new CodeQualityRule(rule.getName(), newRule, true, rule.isAdvanced());
-			} else {
-				rule = new CodeQualityRule(newName, newRule, false, false);
-			}
+		String newRule = getJavascriptIfStatementString(rawRuleConditions);
 
+		try {
+			// Runs pre validation to try to catch some errors
+			preValidateJavascriptCode(newRule);
+		} catch (ScriptException e) {
+			editRulePopup.showMessage("The rule provided has an invalid format. Cannot save it.");
+			return;
+		}
+		
+		if (!validadeDefaultRuleThresholds(rule.getName(), newRule)) {
```

Comment:
does this validate the thresholds for every rule? it isn't very clear that (if) this only applies to default rules :/

Reply:
hummm, eu poderia fazer
`if (rule.isDefault() && !validadeDefaultRuleThresholds(rule.getName(), newRule))`, talvez ficasse mais claro então.
Mas sim, basicamente como está, toda regra entra nessa validação. Mas o método é smart cookie e só valida o que temos definido como default.

Reply:
mesmo que seja redundante, concordo em adicionar mais essa condição ao if só para ficar mais claro :)